### ISSUE TITLE
cli.StringSliceFlag and default value

### ISSUE THREAD

**Issue by vieux** [codegangsta/cli#160]
Hello,
I define my flag as follow:

```go
flFilter = cli.StringSliceFlag{
        Name:  "filter",
        Value: &cli.StringSlice{"flag1", "flag2", "flag3"},
}
```

If I start my binary without arguments, `c.StringSlice("filter")` returns `["flag1", "flag2", "flag3"]` as expected.
If I start my binary with `--filter flag4`,  `c.StringSlice("filter")` returns `["flag1", "flag2", "flag3", "flag4"]`. I find this behaviour odd, I expected it to erase the default value.
Even worst, if I start my binary with `--filter flag1`,  `c.StringSlice("filter")` returns `["flag1", "flag2", "flag3", "flag1"]`.

Is this the expected behavior ? I know I have 3 default value so I can do `c.StringSlice("filter")[3:]` if set, but still it's not perfect.

**Comment by jszwedko** [comment]
Mmm this definitely feels unexpected to me.

**Comment by icexin** [comment]
EnvVar of StringSlice also has this behavior.

**Comment by huguesalary** [comment]
I also have this issue and also feels this behavior is completely unexpected. Is there any plans on fixing this?

I'll see if I can take a stab at it and do a PR too.

**Comment by jszwedko** [comment]
I'd be happy to review a PR for this @huguesalary !

**Comment by jinhao** [comment]
I also hava this issue, expect to change this behavior.

**Comment by jszwedko** [comment]
Fixed by #392 (will be released is v2, but you can use that branch for now).

**Comment by sumittal** [comment]
Has this change merged to the master branch?
